tcms/testruns/tests/test_admin.py

```diff
@@ -91,3 +96,33 @@ def test_changelist_view_icon_column(self):
             """
             ),
         )
+
+    def test_test_execution_statuses_can_be_deleted_but_one_must_remain(self):
+
+        for exe_status in self.negative[1:]:
+            response = self.client.get(
+                reverse("admin:testruns_testexecutionstatus_delete", args=[exe_status])
+            )
+            self.assertContains(response, _("Yes, I'm sure"))
+
+            response = self.client.post(
+                reverse("admin:testruns_testexecutionstatus_delete", args=[exe_status]),
+                {"post": "yes"},
+                follow=True,
+            )
+            self.assertRedirects(
+                response, reverse("admin:testruns_testexecutionstatus_changelist")
+            )
+            self.assertEqual(
+                TestExecutionStatus.objects.filter(pk=exe_status).exists(), False
+            )
+        x = self.negative[0]
+        response = self.client.get(
+            reverse("admin:testruns_testexecutionstatus_delete", args=[x]), follow=True
+        )
+        self.assertContains(
+            response, _("1 negative, 1 neutral & 1 positive status required!"), html=True
```

Comment:
FYI `html=True` only makes a difference if what you search for is an HTML tag:
https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.SimpleTestCase.assertContains

In this case it doesn't do anything.